wallet/wallet.go

```diff
@@ -206,6 +213,98 @@ func (w *SingleAddressWallet) FundTransaction(cs consensus.State, txn *types.Tra
 	return toSign, nil
 }
 
+// Split returns a transaction that splits the wallet in the given number of
+// outputs with given amount. The transaction is funded but not signed.
+//
+// NOTE: split needs to use a minimal set of inputs and therefore does not reuse
+// the fund logic which randomizes the unspent transaction outputs used to fund
+// the transaction
+func (w *SingleAddressWallet) Split(cs consensus.State, outputs int, amount, feePerByte types.Currency, pool []types.Transaction) (types.Transaction, []types.OutputID, error) {
+	// prepare all outputs
+	var txn types.Transaction
+	for i := 0; i < int(outputs); i++ {
```

Comment:
I considered preallocating the outputs for `amount+1` here (to accommodate the change output) but preferred simply appending over the added complexity of overwriting the output value with the change or removing it if the change is 0. Willing to update though if you all don't agree.